src/controller/UserController.ts

```diff
@@ -2,78 +2,105 @@ import { Request, Response } from 'express';
+     * Cria um hash da senha e salva o novo usuário no banco de dados.
+     *
+     * @param req - Objeto de requisição do Express contendo os dados do usuário (nome, email, senha, confirmPassword).
+     * @param res - Objeto de resposta HTTP do Express.
+     * @returns Uma resposta HTTP com status 201 e os dados do usuário criado, ou um erro.
      */
     async createUser(req: Request, res: Response): Promise<Response> {
         const { name, email, password, confirmPassword } = req.body;
 
         try {
-            // Validação do tamanho da senha
             if (password.length < 8) {
                 throw new CustomError(
                     'A senha deve conter pelo menos 8 caracteres',
                     400
                 );
             }
 
-            // Validação de senhas correspondentes
             if (password !== confirmPassword) {
                 throw new CustomError('As senhas não coincidem', 400);
             }
 
-            // Validação do email
             const existingUser = await this.userRepository.findByEmail(email);
             if (existingUser) {
                 throw new CustomError('Email já cadastrado', 400);
             }
 
-            // Cria o hash da senha
             const hashedPassword = await hashPassword(password);
-
-            // Cria o novo usuário
             const newUser = this.userRepository.create({
                 name,
                 email,
                 password: hashedPassword
             });
 
-            // Salva o novo usuário no banco de dados
             await this.userRepository.save(newUser);
 
-            // Retorna uma resposta de sucesso
             return res.status(201).json({
                 message: 'Usuário criado com sucesso',
                 user: { id: newUser.id, email: newUser.email }
             });
         } catch (error) {
-            // Tratamento de erros no processo de criação de usuário
             if (error instanceof CustomError) {
                 return res
                     .status(error.status)
                     .json({ message: error.message });
             }
-
-            // Tratamento de erro desconhecido
             return res
                 .status(500)
                 .json({ message: 'Erro desconhecido ao criar usuário' });
         }
     }
+
+    /**
+     * Retorna os dados do usuário logado.
+     *
+     * Este método busca o usuário logado com base no `userId` presente no payload do token JWT.
+     *
+     * @param req - Objeto de requisição do Express contendo o token JWT decodificado em `req.user`.
+     * @param res - Objeto de resposta HTTP do Express.
+     * @returns Uma resposta HTTP com os dados do usuário logado ou uma mensagem de erro.
+     */
+    async getLoggedUser(req: Request, res: Response): Promise<Response> {
+        try {
+            const userId = (req.user as JwtPayloadCustom).userId;
+            console.log('Fetching user with ID:', userId);
```

Comment:
Console.log e console.error, torna nossa aplicação mais lenta e não terá informação nenhuma a agregar para o front-end, pois só será visível no lugar que estiver hosteando.

Vou até deixar um link aqui de um fórum de uma publicação sobre o porque não deixar console.logs no código. [Forum](https://medium.com/@priyanklad52/why-you-should-remove-console-logs-from-production-code-a-guide-for-next-js-developers-36dee52f5e21)